tests/foreman/cli/test_import.py

```diff
@@ -539,11 +544,204 @@ def test_import_users_recovery(self, test_data):
             ssh.command(u'ls ${HOME}/.transition_data/users*').stdout[:-1]
         )
         for record in transition_data:
-            self.assertNotEqual(User.info({'id': record['sat6']}).stdout, '')
+            self.assertEqual(
+                User.info({'id': record['sat6']}).return_code, 0
+            )
         Import.user({'csv-file': files['users'], 'delete': True})
         # do the cleanup
         ssh.command(u'rm -rf {}'.format(' '.join(pwdfiles)))
 
+    @data(*import_org_data())
+    def test_import_host_collections_default(self, test_data):
+        """@test: Import all System Groups from the default data set
+        (predefined source) as the Host Collections.
+
+        @feature: Import Host-Collections
+
+        @assert: 3 Host Collections created
+
+        """
+        files = dict(self.default_dataset[1])
+        files['users'] = update_csv_values(
+            files['users'],
+            test_data,
+            self.default_dataset[0]
+        )
+        files['system-groups'] = update_csv_values(
+            files['system-groups'],
+            test_data,
+            self.default_dataset[0]
+        )
+        # import the prerequisities
+        self.assertEqual(
+            Import.organization({'csv-file': files['users']}).return_code, 0
+        )
+        self.assertEqual(
+            Import.host_collection(
+                {'csv-file': files['system-groups']}
+            ).return_code, 0
+        )
+        transition_data = csv_to_dataset(
+            ssh.command(
+                u'ls ${HOME}/.transition_data/organizations*'
+            ).stdout[:-1]
+        )
+        # now to check whether the all HC from csv appeared in satellite
+        imp_orgs = csv_to_dataset([files['users']])
+        [imp_org.update(
+            {'sat6': tr['sat6']}
+        ) for tr in transition_data for imp_org in imp_orgs
+            if tr['sat5'] == imp_org['organization_id']]
```

Comment:
This is funky. I *think* this equates the following:

```python
        [
            imp_org.update({'sat6': tr['sat6']})
            for tr in transition_data for imp_org in imp_orgs
            if tr['sat5'] == imp_org['organization_id']
        ]
```

The above is a mis-use of list comprehensions. You're not doing anything with
the generated list. I *think* the above equates to this nested loop:

```python
        for imp_org in imp_orgs:
            for tr in transition_data:
                if tr['sat5'] == imp_org['organization_id']
                    imp_org.update({'sat6': tr['sat6']})
```

Assuming it does, you can simplify things by using `itertools.product`:

```python
        for imp_org, tr in product(imp_orgs, transition_data):
            if tr['sat5'] == imp_org['organization_id']
                imp_org.update({'sat6': tr['sat6']})
```

An example of `itertools.product`:

```python
>>> from itertools import product
>>> for first, second in product([1, 2, 3], [1, 5, 9]):
...     (first, second)
... 
(1, 1)
(1, 5)
(1, 9)
(2, 1)
(2, 5)
(2, 9)
(3, 1)
(3, 5)
(3, 9)
```